asllib/carpenter/README.md

```diff
@@ -0,0 +1,47 @@
+# ASL Carpenter
+
+Carpenter is an AST generator for ASL, working in two different modes:
+
+1. Enumeration of all ASTs by increasing size;
+2. Random generation of ASTs following type-checking constraints.
+
+Both modes support constraints on syntax nodes available, and on execution or
+type-checking rules used by ASLRef to execute it.
+
+It can also serve as a fuzzing tool, or a mass execution tool.
+
+## Installation
+
+Dependencies:
+```bash
+opam install dune qcheck-core zarith re feat fix logs
+```
+
+And herdtools7 (see [herdtools7's readme](../../README.md)).
+
+Then:
+```bash
+dune build carpenter
```

Comment:
I just realised this need to be run from `asllib`, might be worth making it clear here.